### DISCUSSION TITLE
Hello! Can it grab history/marked for later/

### DISCUSSION THREAD

**Discussion by moniquemolina2002** [nianeyna/ao3downloader#127] [General]
Hello! Thank you so much making this! Is there a way for it to download from my "marked for later"? or History? and it would go through each page?

**Comment by nianeyna** [comment]
Yep! There is a specific option to download your marked for later list *and* mark everything in it as read. Or, if you don't want to mark things as read, just use the normal "a" option and input the direct link to your marked for later list (or history) and it will download everything there. Either option will require you to input your login information in order to work properly (since your history is private to you). And yes, it will get all pages.

**Comment by moniquemolina2002** [comment]
Hmm I tried it with l and put it the link to my history and it didn’t work for some reason ! Is it in another area ? I checked the issues and the menu but maybe I misunderstood something , could be I terrible and ask you to point me to the right direction? I’m so sorry! You’ve already done so much !On Feb 19, 2024, at 10:07 PM, nianeyna ***@***.***> wrote:﻿
Yep! There is a specific option to download your marked for later list and mark everything in it as read. Or, if you don't want to mark things as read, just use the normal "a" option and input the direct link to your marked for later list (or history) and it will download everything there. Either option will require you to input your login information in order to work properly (since your history is private to you). And yes, it will get all pages.

**Comment by nianeyna** [comment]
Hm, when you say it "didn't work" what do you mean? did you get an error message? or did it just not do anything at all?

**Comment by moniquemolina2002** [comment]
Hi! It says this and I’m not sure what it means ! (This is for the mark for later)It worked fine with the bookmarks though
![image0](https://github.com/nianeyna/ao3downloader/assets/71778632/03b8abf6-0181-46e7-b452-74a95e700c93)

**Reply by nianeyna** [reply]
This looks like it might be a bug - can you check the log file (menu option "v") and see if there are any error messages?

**Comment by moniquemolina2002** [comment]
Also is there any way to get the summary when it does the csv with links ?

**Reply by nianeyna** [reply]
It *should* be putting the summary in the csv - if it's not doing that, I'm not sure why.

**Comment by moniquemolina2002** [comment]
It will say list not complete for some reason, . Sorry also- is there a way to have it just download new links if I’ve updated my bookmarks or does it have to download everything all over again?

**Reply by nianeyna** [reply]
Not exactly, but you can find the page your most recently downloaded bookmark is on and tell it to stop on that page. (Unfortunately the bookmarks page doesn't let you filter by date bookmarked last I checked, otherwise you could just do that)

**Comment by moniquemolina2002** [comment]
Weird! I’ll send you the file I got and also the logs ! Would it help if I gave you the links and even log in and you looked ?On Feb 20, 2024, at 3:45 PM, nianeyna ***@***.***> wrote:﻿
It should be putting the summary in the csv - if it's not doing that, I'm not sure why.

**Comment by moniquemolina2002** [comment]
Like here just the bookmarks (the mark for later is thousands ) https://archiveofourown.org/users/sakurablossomgirl16/bookmarksOn Feb 20, 2024, at 4:22 PM, Monica ***@***.***> wrote:﻿Weird! I’ll send you the file I got and also the logs ! Would it help if I gave you the links and even log in and you looked ?On Feb 20, 2024, at 3:45 PM, nianeyna ***@***.***> wrote:﻿
It should be putting the summary in the csv - if it's not doing that, I'm not sure why.

**Comment by nianeyna** [comment]
PLEASE do not give me your ao3 login information lol. What you could do however is go to page 80 of your history (or marked for later, whichever you were downloading when it failed) and right-click somewhere on the page. Select the option that says something like "View Page Source". In the window that comes up when you click on that, copy-paste the entire contents of the page and save it as a text file (so in TextEdit, not a word processor). Then send that as well as your log.jsonl file to me via email (nianeyna@gmail.com)

**Comment by nianeyna** [comment]
Ok, I suspect I know what this is, and I just pushed a change that (among other things) I hope will fix it. Please try downloading the newest version of the code and try again!

(Also, I see the summaries just fine in the csvs you sent me?)
